src/jit/gentree.h

```diff
@@ -4845,6 +4856,86 @@ struct GenTreePutArgStk : public GenTreeUnOp
 #endif
 };
 
+#ifdef _TARGET_ARM_
+// Represent the struct argument: value is splitted in register(s) and stack
+struct GenTreePutArgStruct : public GenTreePutArgStk
+{
+    unsigned gtNumRegs;
+    unsigned gtIsHfa;
+
+    GenTreePutArgStruct(GenTreePtr op1,
+                        unsigned slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(unsigned numSlots),
+                        unsigned     numRegs,
+                        bool         isHfa,
+                        bool         putIncomingArgArea = false,
+                        GenTreeCall* callNode           = nullptr)
+        : GenTreePutArgStk(GT_PUTARG_STRUCT,
+                           TYP_STRUCT,
+                           op1,
+                           slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(numSlots),
+                           putIncomingArgArea,
+                           callNode)
+        , gtNumRegs(numRegs)
+        , gtIsHfa(isHfa)
+    {
+    }
+
+    // not consider HFA yet
+    regNumber gtOtherRegs[MAX_REG_ARG - 1];
+
+    //---------------------------------------------------------------------------
+    // GetRegNumByIdx: get ith register allocated to this struct argument.
+    //
+    // Arguments:
+    //     idx   -   index of the struct
+    //
+    // Return Value:
+    //     Return regNumber of ith register of this struct argument
+    //
+    regNumber GetRegNumByIdx(unsigned idx) const
```

Comment:
IMO there are a number of existing names that aren't ideal, but I'm OK with being consistent or switching.